libsolidity/formal/SMTChecker.cpp

```diff
@@ -750,6 +750,18 @@ bool SMTChecker::visit(MemberAccess const& _memberAccess)
 		defineGlobalVariable(accessedName + "." + _memberAccess.memberName(), _memberAccess);
 		return false;
 	}
+	else if (accessType->category() == Type::Category::Enum)
```

Comment:
Shouldn't that be `exprType`? Otherwise this would trigger for
```
contract C {
enum X {A, B}
struct S { X x; }
S s;
function f() public pure {
  s.x; // MemberAccess whose resulting type is enum.
}
}

Reply:
Weird - you already mentioned that here: https://github.com/ethereum/solidity/pull/6198#discussion_r263046251

Reply:
`exprType` has type `TypeType` in that case, since it is the type of an `EnumDefinition`.
I added a testcase very similar to that one. The current code checks if the `_memberAccess.expression()` is an `EnumDefinition`, and only if that's the case retrieves the numerical value. Otherwise it just creates a symbolic for `_memberAccess`.

Reply:
Same as above: Please do checks based on the type of the expression and not the resulting value.